providers/linode/linodeProvider.go

```diff
@@ -277,7 +277,7 @@ func toReq(dc *models.DomainConfig, rc *models.RecordConfig) (*recordEditRequest
 		Type:     rc.Type,
 		Name:     rc.GetLabel(),
 		Target:   rc.GetTargetField(),
-		TTL:      int(rc.TTL),
+		TTL:      int(rc.TTL.Value()),
```

Comment:
This is what fully fixes #2440:
- When no TTL is set, this evaluates to `models.DefaultTTL` (300).
- When `DefaultTTL(0)` is used explicitly, this evaluates to 0.